liquibase-core/src/main/java/liquibase/sqlgenerator/core/CreateProcedureGenerator.java

```diff
@@ -70,15 +70,11 @@ public Sql[] generateSql(CreateProcedureStatement statement, Database database,
             }
             sql.add(new UnparsedSql("if object_id('" + fullyQualifiedName + "', 'p') is null exec ('create procedure " + fullyQualifiedName + " as select 1 a')"));
 
-            StringClauses parsedSql = SqlParser.parse(procedureText, true, true);
-            StringClauses.ClauseIterator clauseIterator = parsedSql.getClauseIterator();
-            Object next = "START";
-            while (next != null && !(next.toString().equalsIgnoreCase("create") || next.toString().equalsIgnoreCase("alter")) && clauseIterator.hasNext()) {
-                next = clauseIterator.nextNonWhitespace();
+            StringClauses parsedProcedureDefinition = SqlParser.parse(procedureText, true, true);
+            if (parsedProcedureDefinition.contains("CREATE") && !(parsedProcedureDefinition.contains("OR") && parsedProcedureDefinition.contains("ALTER"))) {
+                parsedProcedureDefinition.replace("CREATE", "ALTER");
```

Comment:
@achehrynets what if statement has multiple "create" words in it, for example as part of procedure body, is it possible? If it is, won't this line replace all the matches and not only the first one?

Reply:
I added unit tests and found one possible problem which could happen and fixed it